src/components/ReportActionItem/ReportPreview.js

```diff
@@ -0,0 +1,159 @@
+    },
+};
+
+const ReportPreview = (props) => {
+    const reportAmount = CurrencyUtils.convertToDisplayString(props.iouReport.total, props.iouReport.currency);
+    const managerEmail = props.iouReport.managerEmail || '';
+    const managerName = ReportUtils.getDisplayNameForParticipant(managerEmail, true);
+    const isCurrentUserManager = managerEmail === lodashGet(props.session, 'email', null);
+    return (
+        <View style={[styles.chatItemMessage, styles.mt4]}>
+            {_.map(props.action.message, (index) => (
+                <Pressable
+                    key={`ReportPreview-${props.action.reportActionID}-${index}`}
+                    onPress={props.onViewDetailsPressed}
+                    onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
+                    onPressOut={() => ControlSelection.unblock()}
+                    onLongPress={(event) => showContextMenuForReport(event, props.contextMenuAnchor, props.chatReportID, props.action, props.checkIfContextMenuActive)}
+                    style={[styles.flexRow, styles.justifyContentBetween]}
+                    focusable
+                >
+                    <View>
+                        {props.iouReport.hasOutstandingIOU ? (
+                            <Text style={[styles.chatItemMessage, styles.cursorPointer]}>{props.translate('iou.payerOwesAmount', {owner: managerName, amount: reportAmount})}</Text>
+                        ) : (
+                            <View style={[styles.flexRow]}>
+                                <Text style={[styles.chatItemMessage, styles.cursorPointer]}>{props.translate('iou.payerSettled', {owner: managerName, amount: reportAmount})}</Text>
+                                {!props.iouReport.hasOutstandingIOU && (
+                                    <Icon
+                                        style={[styles.ml10]}
+                                        src={Expensicons.Checkmark}
+                                        fill={themeColors.iconSuccessFill}
+                                    />
+                                )}
+                            </View>
+                        )}
+                    </View>
+                    <Icon
+                        src={Expensicons.ArrowRight}
+                        fill={StyleUtils.getIconFillColor(getButtonState(props.isHovered))}
+                    />
+                </Pressable>
+            ))}
+            {isCurrentUserManager && props.iouReport.stateNum === CONST.REPORT.STATE_NUM.PROCESSING && (
+                <Button
+                    style={[styles.requestPreviewBox]}
+                    onPress={() => {
+                        Navigation.navigate(ROUTES.getIouDetailsRoute(props.chatReportID, props.iouReportID));
+                    }}
+                    onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
+                    onPressOut={() => ControlSelection.unblock()}
```

Comment:
similar to the other comment, let's make sure we spin up an issue to use hooks here like `useCallback` or `useMemo` not sure 100% which one is correct here.